tests/qchem/test_hartree_fock.py

```diff
@@ -103,6 +103,48 @@ def test_hf_energy(symbols, geometry, charge, basis_name, e_ref):
     assert np.allclose(e, e_ref)
 
 
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "alpha", "charge", "basis_name", "e_ref"),
+    [
+        (
+            ["H", "H"],
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad=True),
+            [
+                np.array([3.42525091, 0.62391373, 0.1688554], requires_grad=True),
+                np.array([3.42525091, 0.62391373, 0.1688554], requires_grad=True),
+            ],
+            0,
+            "sto-3g",
+            # HF energy computed with pyscf using scf.hf.SCF(mol_pyscf).kernel()
+            np.array([-1.06599931664376]),
+        ),
+        (
+            ["H", "H", "H"],
+            np.array([[0.0, 1.0, 0.0], [0.0, 0.0, 0.0], [1.0, 0.0, 0.0]], requires_grad=True),
+            [
+                np.array([18.73113696, 2.82539437, 0.64012169], requires_grad=True),
+                np.array([0.16127776], requires_grad=True),
+                np.array([18.73113696, 2.82539437, 0.64012169], requires_grad=True),
+                np.array([0.16127776], requires_grad=True),
+                np.array([18.73113696, 2.82539437, 0.64012169], requires_grad=True),
+                np.array([0.16127776], requires_grad=True),
+            ],
+            1,
+            "6-31g",
+            # HF energy computed with pyscf using scf.hf.SCF(mol_pyscf).kernel()
+            np.array([-1.11631458846075]),
+        ),
+    ],
+)
+def test_hf_energy_diff(symbols, geometry, alpha, charge, basis_name, e_ref):
```

Comment:
I think we should also have a test for the dipole moment calculation with `6-31g`.

Reply:
Added a test for constructing dipole moment when the basis parameters are differentiable.